sdccc/src/test/java/com/draeger/medical/sdccc/manipulation/precondition/impl/ConditionalPreconditionsTest.java

```diff
@@ -651,6 +657,535 @@ void testDescriptionModificationModificationNoDescriptors() {
+            } else {
+                presenceMap.put(handle, false);
+                return ResponseTypes.Result.RESULT_SUCCESS;
+            }
+        });
+        when(mockManipulations.triggerDescriptorUpdate(anyString()))
+                .thenAnswer((Answer<ResponseTypes.Result>) invokation -> {
+                    updateMap.put(invokation.getArgument(0), true);
+                    return ResponseTypes.Result.RESULT_SUCCESS;
+                });
+        when(testClient.getSdcRemoteDevice().getMdibAccess().getDescriptor(anyString()))
+                .thenAnswer((Answer<Optional<AbstractDescriptor>>) invocation -> {
+                    final String handle = invocation.getArgument(0);
+                    if (presenceMap.get(handle)) {
+                        return Optional.of(mock(AbstractDescriptor.class));
+                    } else {
+                        return Optional.empty();
+                    }
+                });
+
+        // when
+        assertTrue(
+                ConditionalPreconditions.DescriptionModificationMdsDescriptorPrecondition.manipulation(testInjector));
+
+        // then
+        final var insertCaptor = ArgumentCaptor.forClass(String.class);
+        final var removeCaptor = ArgumentCaptor.forClass(String.class);
+        final var updateCaptor = ArgumentCaptor.forClass(String.class);
+        verify(mockManipulations, times(1)).getRemovableDescriptorsOfClass(MdsDescriptor.class);
+        verify(mockManipulations, times(2)).insertDescriptor(insertCaptor.capture());
+        verify(mockManipulations, times(2)).removeDescriptor(removeCaptor.capture());
+        verify(mockManipulations, times(2)).triggerDescriptorUpdate(updateCaptor.capture());
+
+        assertEquals(descriptor1Handle, updateCaptor.getAllValues().get(0));
+        assertEquals(descriptor2Handle, updateCaptor.getAllValues().get(1));
+        assertEquals(descriptor1Handle, removeCaptor.getAllValues().get(0));
+        assertEquals(descriptor2Handle, removeCaptor.getAllValues().get(1));
+        assertEquals(descriptor1Handle, insertCaptor.getAllValues().get(0));
+        assertEquals(descriptor2Handle, insertCaptor.getAllValues().get(1));
+    }
+
+    /**
+     * Tests whether DescriptionModificationMdsDescriptorPrecondition correctly calls manipulations
+     * in case a RemovableDescriptor is present but cannot be updated.
+     */
+    @Test
+    @DisplayName("DescriptionModificationMdsDescriptorPrecondition correctly calls manipulations when a "
+            + "RemovableDescriptor is present but cannot be updated")
+    public void testDescriptionModificationMdsDescriptorPreconditionGoodUpdateFails() {
+
+        // given
+        final var descriptor1Handle = "superHandle1";
+        final var descriptor2Handle = "handle;Süper;2";
+
+        final var presenceMap = new HashMap<>(Map.of(
+                descriptor1Handle, false,
+                descriptor2Handle, true));
+        final var updateMap = new HashMap<>(Map.of(
```

Comment:
This HashMap is updated but otherwise not used for the test case.

Reply:
Removed.